packages/server/lib/remote_states.ts

```diff
@@ -91,7 +91,7 @@ export class RemoteStates {
 
     if (_.isString(urlOrState)) {
       const remoteOrigin = uri.origin(urlOrState)
-      const { subdomain: _unused, ...remoteProps } = cors.parseUrlIntoHostProtocolDomainTldPort(remoteOrigin)
+      const remoteProps = cors.parseUrlIntoHostProtocolDomainTldPort(remoteOrigin)
```

Comment:
Without this change, checking same-origin of something like www.google.com would always fail since the www was getting dropped. We need to keep subdomain.